src/aiidalab_qe/common/mixins.py

```diff
@@ -0,0 +1,91 @@
+import typing as t
+
+import traitlets as tl
+
+from aiida import orm
+from aiida_quantumespresso.data.hubbard_structure import HubbardStructureData
+
+T = t.TypeVar("T")
+
+
+class MissingMixinError(Exception):
+    """Raised when no mixin is found in a class definition."""
+
+    def __init__(self, *args: object) -> None:
+        super().__init__("expected at least one mixin in class definion", *args)
+
+
+class MetaMixinTraitsRegister(tl.MetaHasTraits):
+    """A metaclass to register traits from `HasTraits`-subclassed mixins.
+
+    The metaclass removes the `HasTraits` base class from the mixin to
+    avoid MRO conflicts.
+    """
+
+    def __new__(cls, name, bases, classdict):
+        if len(bases) == 1 and not issubclass(bases[0], tl.HasTraits):
+            raise MissingMixinError()
+        for base in bases:
+            if issubclass(base, tl.HasTraits):
+                for name, trait in base.class_traits().items():
+                    if name not in classdict:
+                        classdict[name] = trait
+        bases = tuple(filter(lambda base: base is not tl.HasTraits, bases))
```

Comment:
I think you want to remove the classes which are the subclass of `tl.HasTraits`.
```suggestion
        bases = tuple(filter(lambda base: not issubclass(base, tl.HasTraits), bases))
```